### PR TITLE
(feat) O3-3443 Command in the OpenMRS CLI to package up configurations

### PR DESCRIPTION
**Pull request by suubi-joshua**
# Requirements

- [x] This PR has a title that briefly describes the work done including the ticket number. Ensure your PR title includes a [conventional commit](https://o3-docs.openmrs.org/docs/frontend-modules/contributing#contributing-guidelines) label (such as `feat`, `fix`, or `chore`, among others). See existing PR titles for inspiration.

## For changes to apps

- [ ] My work conforms to the [**O3 Styleguide**](https://om.rs/styleguide) and [**design documentation**](https://om.rs/o3ui).

## If applicable

- [ ] My work includes tests or is validated by existing tests.
- [ ] I have updated the [esm-framework mock](https://github.com/openmrs/openmrs-esm-core/blob/main/packages/framework/esm-framework/mock.tsx) to reflect any API changes I have made.

## Summary
This script automatically merges configuration files from multiple directories into a single file. It simplifies managing app-specific configurations and reduces manual maintenance.

## Related Issue

https://openmrs.atlassian.net/browse/O3-3443

### CODE REVIEW

**Comment by gracepotma** [comment]
@wikumChamith does this look to you like what @ibacher had in mind for https://openmrs.atlassian.net/browse/O3-3443?

**Comment by ibacher** [comment]
No, it does not. I think this does something else. @suubi-joshua How can I test this? What does it do? Why is there anything about config schemas in this? The goal of the ticket was solely to allow a user to specify some JSON files and merge those into a single thing.

**Comment by suubi-joshua** [comment]
Thanks for the clarity @ibacher. The script i have provided allows user to specify directories pointing to ts files containing custom configSchemas then merges them into one file. I assumed that configs are in typescript format just like the [here ](https://github.com/openmrs/openmrs-esm-core/blob/main/packages/apps/esm-login-app/src/config-schema.ts)configschemas not json files. When working proper this script writes the combined configs as an object in a ts file (working on testing this in cli). 

I am working on the testing it in the CLI.

File: packages/framework/esm-framework/docs/interfaces/WorkspaceContainerProps.md [hunk 1]
```diff
@@ -19,7 +19,9 @@
 
 #### Defined in
 
+=======
 [packages/framework/esm-styleguide/src/workspaces/container/workspace-container.component.tsx:11](https://github.com/openmrs/openmrs-esm-core/blob/main/packages/framework/esm-styleguide/src/workspaces/container/workspace-container.component.tsx#L11)
+>>>>>>> upstream/main
```
[the remark this answers is not in the document]
Comment (suubi-joshua) [reply packages/framework/esm-framework/docs/interfaces/WorkspaceContainerProps.md:24]: This came after fetching the changes from upstream and merged them.

**Comment by suubi-joshua** [comment]
> No, it does not. I think this does something else. @suubi-joshua How can I test this? What does it do? Why is there anything about config schemas in this? The goal of the ticket was solely to allow a user to specify some JSON files and merge those into a single thing.

@ibacher @wikumChamith With testing the command to use in the cli is `yarn merge-config -d <directories to config files> -o <directory to output file>

**Comment by ibacher** [comment]
@suubi-joshua I think that it's preferable that this all gets wrapped up into the "assemble" command. I don't not want to add more steps that implementers need to run, just more options to the steps they can run.

**Comment by ibacher** [comment]
Also, it's better to allow users to specify files rather than directories. We can always make `--config-file configs/*` work, but we cannot make `--config-file my-config.json --config-file my-config-2.json` work.

**Comment by ibacher** [comment]
Similarly, the output file name shouldn't be required to be specified, but have a sane default.

**Comment by gracepotma** [comment]
(Removed Wikum as he explained to me this is outside his familiarity area at this time.)

File: packages/tooling/openmrs/src/commands/assemble.ts [hunk 2]
```diff
@@ -17,6 +17,7 @@ export interface AssembleArgs {
   target: string;
   mode: string;
   config: Array<string>;
+  myConfigs: Array<string>;
```
Comment (ibacher) [inline packages/tooling/openmrs/src/commands/assemble.ts:20]: Better naming suggestion:

```suggestion
  configFiles: Array<string>;
```

File: packages/tooling/openmrs/src/commands/assemble.ts [hunk 3]
```diff
@@ -303,5 +331,9 @@ export async function runAssemble(args: AssembleArgs) {
     await writeFile(resolve(args.target, 'spa-assemble-config.json'), JSON.stringify(versionManifest), 'utf8');
   }
 
+  if(args.mode === 'myConfigs'){
+    await writeFile(resolve(args.target, 'My-Merged-configs.json'), JSON.stringify(config), 'utf8');
```
Comment (ibacher) [inline packages/tooling/openmrs/src/commands/assemble.ts:335]: This filename should be `spa-config.json`

File: packages/tooling/openmrs/src/commands/assemble.ts (hunk 2 above)
Comment (suubi-joshua) [reply packages/tooling/openmrs/src/commands/assemble.ts:20]: Okay will work on this

File: packages/tooling/openmrs/src/commands/assemble.ts (hunk 3 above)
Comment (suubi-joshua) [reply packages/tooling/openmrs/src/commands/assemble.ts:336]: Okay

File: packages/tooling/openmrs/src/cli.ts [hunk 4]
```diff
@@ -289,10 +289,10 @@ yargs.command(
...
       .option('mode', {
-        choices: ['config', 'survey'],
+        choices: ['config', 'configFiles', 'survey'],
         default: 'survey',
         description:
-          'The source of the frontend modules to assemble. `config` uses a configuration file specified via `--config`. `survey` starts an interactive command-line survey.',
+          'The source of the frontend modules to assemble. `config` uses a configuration file specified via `--config`. `configFiles` packages up configurations from file specified via `--config-file`. `survey` starts an interactive command-line survey.',
         type: 'string',
       }),
   (args) => runCommand('runAssemble', args),
```
Comment (suubi-joshua) [inline packages/tooling/openmrs/src/cli.ts:298]: Hello @ibacher I am facing a blocker. After making the changes I want to test them but there is a persistent error in` cli.ts` file on the `args,` somehow the new argument of `configFiles` is missing from the `args` in the `cli `yet its exported in the `AssembleArgs interface`.  Also when I edit the mode option to put another choice ie  `configFiles` it does not seem to get added (How do I know this, when i run `yarn assemble --help` my edits for choices dont show even for the description)

**Comment by brandones** [comment]
@ibacher @suubi-joshua What's the status of this, how should we move it forward?

**Comment by ibacher** [comment]
I'm going to close this as superceded. Sorry @suubi-joshua! Didn't mean to ignore your work.

**Pull request closed without merging**